### PR TITLE
feat: Create Consequence Services

### PR DESCRIPTION
**Pull request by alainajaffer**
As per wireframes it should be similar to stops with two drop downs. The stops dropdown should not populate until the service is selected from the service dropdown then a user can select a value from the stops dropdown. When a service is removed all the stops associated should be removed.

### CODE REVIEW

**Comment by lj-kpmg** [comment]
I think this page needs the direction option according to AC7

File: site/schemas/consequence.schema.ts [hunk 1]
```diff
@@ -51,12 +52,26 @@ export type Stop = z.infer<typeof stopSchema>;
...
+    operatorShortName: z.string(),
+    destination: z.string(),
+    origin: z.string(),
+});
+
 export const servicesConsequenceSchema = z.object({
     ...baseConsequence,
     consequenceType: z.literal("services", setZodDefaultError("Select a consequence type")),
-    services: z.array(z.object({ id: z.string(), name: z.string() })),
+    stops: z.array(stopSchema).optional(),
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/schemas/consequence.schema.ts]: done

File: site/pages/create-consequence-services.page.tsx [hunk 2]
```diff
@@ -0,0 +1,522 @@
...
+            if (selectedService) {
+                const searchApiUrl = `${API_BASE_URL}services/${selectedService.id}/stops`;
+                const res = await fetch(searchApiUrl, { method: "GET" });
+                const data: Stop[] = z.array(stopSchema).parse(await res.json());
+                if (data) {
+                    const dataWithServiceId = data.map((stop) => ({
+                        ...stop,
+                        ...(selectedService.id && { serviceId: selectedService.id }),
+                    }));
+                    setStopOptions(dataWithServiceId);
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/pages/create-consequence-services.page.tsx]: done

File: site/pages/create-consequence-services.page.tsx [hunk 3]
```diff
@@ -0,0 +1,522 @@
...
+                    }),
+                    services,
+                },
+                errors: pageState.errors,
+            });
+
+            services.splice(index, 1);
+        }
+        setSelectedService(null);
+        setStopOptions([]);
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/pages/create-consequence-services.page.tsx]: done

File: site/pages/create-consequence-services.page.tsx [hunk 4]
```diff
@@ -0,0 +1,522 @@
...
+            return (
+                a.lineName.localeCompare(b.lineName) ||
+                a.origin.localeCompare(b.origin) ||
+                a.destination.localeCompare(b.destination) ||
+                a.operatorShortName.localeCompare(b.operatorShortName)
+            );
+        });
+    };
+
+    const removeService = (e: SyntheticEvent, index: number) => {
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/pages/create-consequence-services.page.tsx]: done

File: site/pages/create-consequence-stops.page.tsx [hunk 5]
```diff
@@ -32,6 +32,18 @@ import { getStateUpdater } from "../utils/formUtils";
 const title = "Create Consequence Stops";
 const description = "Create Consequence Stops page for the Create Transport Disruptions Service";
 
+export const getStopLabel = (stop: Stop) => {
```
[the remark this answers is not in the document]
Comment (alainajaffer) [reply site/pages/create-consequence-stops.page.tsx]: done

File: site/pages/create-consequence-services.page.tsx [hunk 6]
```diff
@@ -0,0 +1,542 @@
...
+
+    if (typeCookie) {
+        const previousConsequenceInformation = typeOfConsequenceSchema.safeParse(JSON.parse(typeCookie));
+
+        if (previousConsequenceInformation.success) {
+            previousConsequenceInformationData = previousConsequenceInformation.data;
+        }
+    }
+
+    inputs = getPageStateFromCookies<StopsConsequence>(dataCookie, errorCookie, stopsConsequenceSchema);
```
Comment (lj-kpmg) [inline site/pages/create-consequence-services.page.tsx:528]: Anywhere with `StopsConsequence` should be replaced with `ServicesConsequence` and same with `stopsConsequenceSchema` to `servicesConsequenceSchema`

Comment (alainajaffer) [reply site/pages/create-consequence-services.page.tsx:528]: done

**Review by lj-kpmg — APPROVED** [review]

**Pull request merged**